### ISSUE TITLE
how to add another layer of data?

### ISSUE THREAD

**Issue by j2l** [saltcorn/leaflet-map#3]
I'd need a second table (with lat, long fields of course) to be visible on the map.

It's a good time to start to learn how to develop a plugin for saltcorn :smile: 

How to add markers from another table?

I tried to start tweaking this plugin locally

BTW, why run() AND renderRows()? These two are very similar.

Thanks

**Comment by j2l** [comment]
@glutamate I now add icons to the plugin, and published this new plugin for saltcorn!
https://github.com/j2l/leaflet-map-icon

Please, please, let me know how to add another table and how to fetch data from REST API to populate this second table.
I'm banging my head on walls :)
Thanks

**Comment by glutamate** [comment]
Thanks, I will look at your new plugin!

It's not so easy to add another table, I think the best bet is to create another view that is configured differently:

1) tableless, so it isn't tied to a table.
2) configuration uses field repeat where in each repeated section you choose a table, lat and long variables

You can ignore the complexities about runMany and renderRows, they are there to make things more efficient if you have repeated displays of a view. In this case I had some dream about a kanban board with a map in each card, used to track lost kittens, but if you only want one map you can just implement run.

I might be able to implement a multi table map soon

**Comment by j2l** [comment]
Thank you!
> I might be able to implement a multi table map soon

Hooo, great!
A "one map, many tables" solution could be valuable for other people too :)

**Comment by j2l** [comment]
And what about an action merging (upserting and rewriting IDs) two tables (having the fields of both) into one?
This way, the action takes care of getting a final table to work with for any view. You could stack this action multiple times to merge more than two tables.
```
 rowsGlobal = rowsTable1.concat(rowsTable2);
  // re-index ids
  uniq = rowsGlobal.filter(function(item, pos) {
    rowsGlobal[pos].id = pos+1; // starts at 1
    return rowsGlobal;
});
```

A link in saltcorn > tables to duplicate a table with fields configured might help :)

What do you think @glutamate ?

**Comment by glutamate** [comment]
multitable map implemented in 0.1.5. 

You need to create a new view, of the "leaflet map multitable" template, in that you can select multiple tables which each has its own lat/long fields and views

I also stole your icon code (Single table maps only) @j2l
